illumos-utils/src/zfs.rs

```diff
@@ -931,6 +945,36 @@ pub fn get_all_omicron_datasets_for_delete() -> anyhow::Result<Vec<String>> {
 mod test {
     use super::*;
 
+    // This test validates that "get_values" at least parses correctly.
+    //
+    // To minimize test setup, we rely on a zfs dataset named "rpool" existing,
+    // but do not modify it within this test.
+    #[cfg(target_os = "illumos")]
+    #[test]
+    fn get_values_of_rpool() {
+        // If the rpool exists, it should have a name.
+        let values = Zfs::get_values("rpool", &["name"; 1], None)
+            .expect("Failed to query rpool type");
+        assert_eq!(values[0], "rpool");
+
+        // We don't really care if any local properties are set, we just don't
+        // want this to throw an error.
+        let _values =
+            Zfs::get_values("rpool", &["name"; 1], Some(PropertySource::Local))
+                .expect("Failed to query rpool type");
```

Comment:
This isn't the *strongest* test in the world, but it does catch the case above, where `-s local` is passed after `all_names` above.

I did this, for now, because it's really light-on-overhead (there's no setup/teardown)